### PR TITLE
fix(packaging/#2813): fix release build on macOS Big Sur

### LINKED ISSUE #2813: Release binary crashes on MacOS 11.0.1
### Versions:
* MacOS: 11.0.1
* node: v15.0.3
* npm: 7.0.14
* esy: 0.6.7
* onivim/oni2: commit 533d1035
* fish shell: 3.1.2
* alacritty: 0.6.0 (04cbf76)

### Bug:
The release binary of Onivim2 crashes immediatelly after execution. The window pops up for just a millisecond.
Interesting is that executing Onivim2 by calling `esy run` works just fine.

### Log:
```bash
$ esy '@release' run -f --checkhealth
dyld: Library not loaded: @rpath/Sparkle.framework/Versions/A/Sparkle
  Referenced from: /Users/bchmnn/Development/oni2/_esy/release/store/i/oni2-fb471e05/bin/Oni2_editor
  Reason: image not found
```
```bash
$ _esy/release/install/bin/Oni2 -f --trace --debug
dyld: Library not loaded: @rpath/Sparkle.framework/Versions/A/Sparkle
  Referenced from: /Users/bchmnn/Development/oni2/./_esy/release/install/bin/Oni2_editor
  Reason: image not found
```
```bash
$ _release/Onivim2.app/Contents/MacOS/Oni2 -f --trace --debug
[DEBUG] [ 0.000s] Oni2.Exception : Recording backtraces
[INFO]  [ 0.000s] Oni2_editor : Starting Onivim 2.0.5.1-nightly (01afc541)
[DEBUG] [ 0.000s] Oni2_editor : Calling App.start
[DEBUG] [ 0.053s] Oni2_editor : Init
[INFO]  [ 0.068s] Oni2_editor : Startup: Trying to change folder to: /Users/bchmnn/Development/oni2
[INFO]  [ 0.068s] Oni2_editor :  - Have read permission
[INFO]  [ 0.069s] Oni2_editor : - Ran chdir
[INFO]  [ 0.069s] Oni2_editor : Successfully changed working directory to: /Users/bchmnn/Development/oni2
[INFO]  [ 0.069s] Oni2_editor : Unsanitized x value: 25
[INFO]  [ 0.069s] Oni2_editor : Unsanitized x value: 0
[INFO]  [ 0.069s] Oni2_editor : Sanitized values from persistence - x: Absolute(0) y: Absolute(25) width: 1440 height: 875
[DEBUG] [ 0.069s] Revery.Core.Window : Starting window creation...
[INFO]  [ 0.069s] Revery.Core.Window : Display count: 1
[INFO]  [ 0.069s] Revery.Core.Window : Display 0:Color LCD - pixelFormat: SDL_PIXELFORMAT_ARGB8888 width: 1440 height: 900 refreshRate: 60 Bounds: x: 0 y: 0 width: 1440 height: 900
[INFO]  [ 0.069s] Revery.Core.Window : Creating window Oni2 width: 1440 height: 875
[INFO]  [ 0.100s] Revery.Core.Window : Window created successfully.
[DEBUG] [ 0.100s] Revery.Core.Window : - Id: 1
[DEBUG] [ 0.101s] Revery.Core.Window : - PixelFormat: SDL_PIXELFORMAT_ARGB8888
[DEBUG] [ 0.101s] Revery.Core.Window : Setting window context
[DEBUG] [ 0.122s] Revery.Core.Window : GL setup. Checking GL version...
[DEBUG] [ 0.122s] Revery.Core.Window : Checking GL vendor...
[DEBUG] [ 0.122s] Revery.Core.Window : Checking GL shading language version...
[INFO]  [ 0.122s] Revery.Core.Window : OpenGL hardware info:
[INFO]  [ 0.122s] Revery.Core.Window :   renderer: Intel(R) Iris(TM) Plus Graphics 655
[INFO]  [ 0.122s] Revery.Core.Window :   version: 4.1 INTEL-16.0.49
[INFO]  [ 0.122s] Revery.Core.Window :   vendor: Intel Inc.
[INFO]  [ 0.122s] Revery.Core.Window :   shadingLanguageVersion: 4.10
[DEBUG] [ 0.122s] Revery.Core.Window : Loading icon from: /Users/bchmnn/Development/oni2/_release/Onivim2.app/Contents/MacOS/logo.png
[ERROR] [ 0.122s] Revery.Core.Window : Error loading icon: can't fopen
[DEBUG] [ 0.122s] Revery.Core.Window : Getting window metrics
[DEBUG] [ 0.122s] Revery.Core.Window : Metrics: { scaledSize = { width = 1440; height = 847 };
  unscaledSize = { width = 1440; height = 847 };
  framebufferSize = { width = 2880; height = 1694 }; devicePixelRatio = 2.;
  scaleFactor = 1.; zoom = 1.; isDirty = false }
[INFO]  [ 0.123s] Revery.Core.Window : Using vsync: Immediate
[DEBUG] [ 0.201s] Oni2_editor : Initializing setup.
[DEBUG] [ 0.201s] Oni2.Core.Setup : Looking for setup configuration at: /Users/bchmnn/Development/oni2/_release/Onivim2.app/Contents/MacOS/setup.json
[WARN]  [ 0.222s] Revery.FontCache : Error loading typeface (skia)
[ERROR] [ 0.225s] Oni2.Exception : Exception (Invalid_argument "result is Error _"):
Raised at file "stdlib.ml", line 30, characters 20-45
Called from file "src/Core/Utility/OptionEx.re" (inlined), line 65, characters 12-15
Called from file "src/Core/FontMeasurementCache.re", line 20, characters 4-352
Called from file "src/Core/Font.re", line 33, characters 6-121
Called from file "src/bin_editor/Oni2_editor.re", line 247, characters 14-36
Called from file "src/Core/App.re", line 199, characters 17-34
Called from file "packages/reason-sdl2/src/sdl2.re", line 269, characters 17-59
Called from file "packages/reason-sdl2/src/sdl2.re", line 269, characters 17-59
Called from file "src/bin_editor/Oni2_editor.re", line 481, characters 11-26
```

### PR DESCRIPTION
**Pull request by brdoney**
Partially fixes #2813 - `esy-macdylibbundler` still needs to be updated for the release builds to work completely on macOS Big Sur.

### CODE REVIEW

File: docs/docs/for-developers/building.md [hunk 1]
```diff
@@ -114,7 +114,9 @@ We use auto formatting tool, you might want to run it before you commit changes
 
 - `esy format`
 
-## Release 
+## Release
+
+> __NOTE:__ On macOS, checkhealth may report that a library is not loaded (e.g. Sparkle). This is normal. To check that libraries are loaded in a generated release build, see the section on macOS below.
```
Comment (bryphe) [inline docs/docs/for-developers/building.md:119]: Thanks for updating the documentation here, @brdoney !

File: scripts/release.js [hunk 2]
```diff
@@ -196,6 +196,21 @@ if (process.platform == "linux") {
     // Remove development plist file
     fs.removeSync(path.join(binaryDirectory, "Info.plist"))
 
+    // The Oni2 and Oni2_Editor binaries can't be symlinks, so replace them with their resolved counterpart
```
Comment (bryphe) [inline scripts/release.js:199]: Nice catch here 👍

**Review by bryphe — APPROVED** [review]
The code change looks great @brdoney - thank you!

As mentioned in #2813 - I just need to validate that this works with our internal code-signing and notarization infrastructure. Once I've verified that - this is good to go.

**Comment by bryphe** [comment]
Tested this locally; looks great @brdoney . Thanks again for your help! 🎉 

This would qualify for our [License Bounty](https://onivim.github.io/docs/for-developers/free-license) - what's a good way to send you the key? Are you on twitter or Discord?

**Comment by brdoney** [comment]
Happy to help! 🎉 

Thank you so much for your patience and all your kind words!

You can reach me through the email listed on GitHub or through Discord - my username there is the same as my GitHub username and I'm on the onivim server as well.

**Comment by bryphe** [comment]
Perfect! I just sent it to the e-mail on your Github profile 👍

**Pull request merged**